meta-luneos/recipes-webos/com.webos.service.mediacontroller/com.webos.service.mediacontroller/0001-com.webos.service.mediacontroller-Make-it-work-gener.patch

```diff
@@ -0,0 +1,169 @@
+From 5a13a97c2112ec58d5171b8cbbe4c85e8570c56d Mon Sep 17 00:00:00 2001
+From: Herrie <Github.com@herrie.org>
+Date: Thu, 21 Dec 2023 12:54:37 +0100
+Subject: [PATCH] com.webos.service.mediacontroller: Make it work generically
+
+Not liking the hardcoded bits from OSE, let's make it more flexible and keep the same functionality
+---
+Upstream-Status: Pending
+
+ CMakeLists.txt              |  6 ++----
+ src/MediaControlService.cpp | 35 ++++++++++++++---------------------
+ src/MediaSessionManager.cpp |  2 +-
+ 3 files changed, 17 insertions(+), 26 deletions(-)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 13f158f..6434b6e 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -45,12 +45,10 @@ if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+ add_definitions(-DUSE_TEST_METHOD)
+ endif()
+ 
+-if(WEBOS_TARGET_MACHINE STREQUAL "raspberrypi4" OR WEBOS_TARGET_MACHINE STREQUAL "raspberrypi4-64" OR WEBOS_TARGET_MACHINE STREQUAL "qemux86" OR WEBOS_TARGET_MACHINE STREQUAL "qemux86-64")
+-add_definitions(-DPLATFORM_RASPBERRYPI4)
+-endif()
+-
+ if(WEBOS_TARGET_MACHINE STREQUAL "sa8155")
+ add_definitions(-DPLATFORM_SA8155)
++else ()
++add_definitions(-DPLATFORM_GENERIC)
+ endif()
+ 
+ webos_add_compiler_flags(ALL -Wall -funwind-tables)
+diff --git a/src/MediaControlService.cpp b/src/MediaControlService.cpp
+index 5faecac..537c8d8 100644
+--- a/src/MediaControlService.cpp
++++ b/src/MediaControlService.cpp
+@@ -150,14 +150,7 @@ bool MediaControlService::onBTAdapterQueryCb(LSHandle *lshandle, LSMessage *mess
+       std::string adapterAddress = adapters[i]["adapterAddress"].asString();
+       PMLOG_INFO(CONST_MODULE_MCS,"%s : adapterName : %s, adapterAddress : %s",
+                                    __FUNCTION__, adapterName.c_str(), adapterAddress.c_str());
+-
+-#if defined(PLATFORM_RASPBERRYPI4)
+-      if(("raspberrypi4 #2" == adapterName) || ("raspberrypi4" == adapterName) ||
+-         ("raspberrypi4-64 #2" == adapterName) || ("raspberrypi4-64" == adapterName) ||
+-         ("qemux86-64 #2" == adapterName) || ("qemux86-64" == adapterName)) {
+-#elif defined(PLATFORM_SA8155)
+-      if ("sa8155 Bluetooth hci0" == adapterName) {
+-#endif
++      if (("" != adapterName) && ("sa8155 Bluetooth hci1" != adapterName)) {
+         deviceSetId = "RSE-L";
+         std::string payload = "{\"adapterAddress\":\"" + adapterAddress + "\",\"subscribe\":true}";
+         PMLOG_INFO(CONST_MODULE_MCS,"%s : payload = %s for first adapter",__FUNCTION__, payload.c_str());
+@@ -170,7 +163,7 @@ bool MediaControlService::onBTAdapterQueryCb(LSHandle *lshandle, LSMessage *mess
+             PMLOG_ERROR(CONST_MODULE_MCS,"%s : LSCall failed for BT device/getStatus", __FUNCTION__);
+         }
+       }
+-      else if ("sa8155 Bluetooth hci1" == adapterName) {
++      else if (("" != adapterName) && ("sa8155 Bluetooth hci1" == adapterName)) {
```

Comment:
unneeded here. If it's "sa8155 Bluetooth hci1", then it's not empty

Reply:
Ah true, I'll change that